src/Vehicle/VehicleBatteryFactGroup.cc

```diff
@@ -123,6 +126,18 @@ void VehicleBatteryFactGroup::_handleHighLatency2(Vehicle* vehicle, mavlink_mess
     group->_setTelemetryAvailable(true);
 }
 
+void VehicleBatteryFactGroup::storeBatteryValue()
+{
+    for(auto i = 0; i < _last_known_consumed.size(); ++i) {
+        _last_known_consumed[i].first = _last_known_consumed[i].second;
```

Comment:
This assumes the batteries in last_known_consumed always have IDs incrementing by one from zero, which could not be the case